packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js

```diff
@@ -162,13 +212,72 @@ describe('licence-summary > route', () => {
     })
 
     it('return value of licenceTypeDisplay is used for licenceTypeStr', async () => {
-      const returnValue = Symbol('return value')
-      licenceTypeDisplay.mockReturnValueOnce(returnValue)
-
+      const mockTypeDisplayValue = Symbol('type display return value')
+      const mockStartTimeValue = Symbol('start time return value')
+      licenceTypeDisplay.mockReturnValueOnce(mockTypeDisplayValue)
+      displayStartTime.mockReturnValueOnce(mockStartTimeValue)
       const result = await getData(mockRequest)
       const ret = result.licenceTypeStr
+      expect(ret).toEqual(mockTypeDisplayValue)
+    })
+  })
+
+  describe('checkNavigation', () => {
+    it('should throw a GetDataRedirect if no licensee first name or last name is found', () => {
+      const permission = { licensee: { firstName: undefined, lastName: undefined } }
+      expect(() => checkNavigation(permission)).toThrow(GetDataRedirect)
+    })
+
+    it('should throw a GetDataRedirect if no date of birth is found', () => {
+      const permission = {
+        licensee: {
+          firstName: 'Barry',
+          lastName: 'Scott',
+          birthDate: undefined
+        }
+      }
+      expect(() => checkNavigation(permission)).toThrow(GetDataRedirect)
+    })
 
-      expect(ret).toEqual(returnValue)
+    it('should throw a GetDataRedirect if no licence start date is found', () => {
+      const permission = {
+        licensee: {
+          firstName: 'Barry',
+          lastName: 'Scott',
+          birthDate: '1946-01-01'
+        },
+        licenceStartDate: undefined
+      }
+      expect(() => checkNavigation(permission)).toThrow(GetDataRedirect)
+    })
+
+    it('should throw a GetDataRedirect if no number of rods or licence type is found', () => {
+      const permission = {
+        licensee: {
+          firstName: 'Barry',
+          lastName: 'Scott',
+          birthDate: '1946-01-01'
+        },
+        licenceStartDate: '2021-07-01',
+        numberOfRods: undefined,
+        licenceType: undefined
+      }
+      expect(() => checkNavigation(permission)).toThrow(GetDataRedirect)
+    })
+
+    it('should throw a GetDataRedirect if no licence length is found', () => {
+      const permission = {
+        licensee: {
+          firstName: 'Barry',
+          lastName: 'Scott',
+          birthDate: '1946-01-01'
+        },
+        licenceStartDate: '2021-07-01',
+        numberOfRods: '3',
+        licenceType: 'Salmon and sea trout',
+        licenceLength: undefined
+      }
+      expect(() => checkNavigation(permission)).toThrow(GetDataRedirect)
```

Comment:
You've added new tests, which is good, but you don't look to be testing the single piece of new code you've put into the getData, which is specifying the locale when determining the licensee's birth date.